src/samplepath.jl

```diff
@@ -0,0 +1,91 @@
+# samplepath.jl
+
+"""
+    abstract type AbstractSamplePath <: Any
+
+subtype: SamplePath.
+
+Its subtype implements a continuous samplepath.
+"""
+abstract type AbstractSamplePath end
+
+"""
+    isincreasing(x::AbstractVector{T})::Bool where T <: Number
+    isincreasing(x::Range{T})::Bool where T <: Number
+
+Tests whether a vector of numbers is strictly increasing. Is internal to
+NonparametricBayesForDiffusions.
+"""
+function isincreasing(x::AbstractVector{T}) where T <: Number
+  for i in 1:length(x)-1
+    x[i+1] <= x[i] && return false
+  end
+  return true
+end
+
+isincreasing(x::Range{T}) where T <: Number = step(x) > 0
+
+"""
+    SamplePath(timeinterval::S, samplevalues::T) where
+        {S<:AbstractVector{Float64}, T<:AbstractVector{Float64}}
+
+ Implements a continuous samplepath where the timeinterval is not necessarily
+ equally spaced. Sample value samplevalues[k] is the value of the process at
+ time timeinterval[k]. timeinterval is an increasing vector. timeinterval and
+ samplevalues are of equal length.
+
+ Constructors:
+ SamplePath(timeinterval, samplevalues) (as above)
+ SamplePath(timeinterval, f::Function) =
+    SamplePath(timeinterval, f.(timeinterval))
+
+ # Examples
+
+```julia
+using PyPlot
+clf()
+t = 0.0:0.1:2.0
+v = sinpi.(t)
+X = SamplePath(t, v)
+plot(X)
+```
+"""
+struct SamplePath{S<:AbstractVector{Float64},
+        T<:AbstractVector{Float64}} <: AbstractSamplePath
+  timeinterval::S
```

Comment:
I have plans to change my SamplePath to allow for abstract time vectors. Why do you not name the elements tt, yy. That makes convergence with Bridge simpler.